### PR TITLE
Add RelativeMaxNumericValidator

### PR DESCRIPTION
**Pull request by vincent-karuri**
PR for issue #130.

### CODE REVIEW

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/FormUtils.java [hunk 1]
```diff
@@ -648,4 +652,61 @@ public JSONArray getSecondaryValues(JSONObject jsonObject, String type) {
 
         return value;
     }
+
+    public static JSONArray fields(JSONObject jsonForm) {
```
Comment (githengi) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/FormUtils.java]: Allow passing the step as a variable, or rename the method

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java [hunk 2]
```diff
@@ -274,4 +279,24 @@ private void addNumericIntegerValidator(JSONObject jsonObject, MaterialEditText
             }
         }
     }
+
+    private void addRelativeNumericIntegerValidator(JSONObject editTextJSONObject, JsonFormFragment formFragment, MaterialEditText editText) throws JSONException {
+        JSONObject numericIntegerObject = editTextJSONObject.optJSONObject(JsonFormConstants.V_NUMERIC_INTEGER);
```
Comment (githengi) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java]: Remove this, should be checked independently by the Numeric Integer validator

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java [hunk 3]
```diff
@@ -274,4 +279,24 @@ private void addNumericIntegerValidator(JSONObject jsonObject, MaterialEditText
...
+        if (numericIntegerObject != null) {
+            String numericValue = numericIntegerObject.optString(JsonFormConstants.VALUE);
+            if (!TextUtils.isEmpty(numericValue) && Boolean.TRUE.toString().equalsIgnoreCase(numericValue)
```
Comment (githengi) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java]: Refactor this to check that the field has defined the dependent field that has the maximum.

Also check that the field is actually defined in the form

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java (hunk 2 above)
Comment (vincent-karuri) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java]: Done.

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/FormUtils.java (hunk 1 above)
Comment (vincent-karuri) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/FormUtils.java]: Done.

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java (hunk 3 above)
Comment (vincent-karuri) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java]: Done.

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java [hunk 4]
```diff
@@ -274,4 +282,25 @@ private void addNumericIntegerValidator(JSONObject jsonObject, MaterialEditText
...
+            String relativeMaxValidationKey = relativeMaxValidationJSONObject.optString(JsonFormConstants.VALUE);
+            JSONObject formJSONObject = new JSONObject(formFragment.getCurrentJsonState());
+            JSONArray formFields = fields(formJSONObject, STEP1);
+            JSONObject relativeMaxFieldJSONObject = getFieldJSONObject(formFields, relativeMaxValidationKey);
+            if (relativeMaxFieldJSONObject != null) {
+                String relativeMaxValidationErrorMsg = relativeMaxValidationJSONObject.optString(JsonFormConstants.ERR);
+                String defaultErrMsg = String.format(DEFAULT_RELATIVE_MAX_VALIDATION_ERR, relativeMaxValidationKey);
+                // add validator
+                editText.addValidator(new RelativeMaxNumericValidator(
+                        relativeMaxValidationErrorMsg == null ? defaultErrMsg : relativeMaxValidationErrorMsg,
```
Comment (githengi) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java]: if the err message was not specified then relativeMaxValidationErrorMsg will be empty string, if you want to check using null the use   
`String relativeMaxValidationErrorMsg = relativeMaxValidationJSONObject.optString(JsonFormConstants.ERR,null);`

Comment (vincent-karuri) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java]: Done.

**Pull request merged**